### PR TITLE
SWP-106 & SWP-82 - multitenancy component

### PR DESCRIPTION
**Pull request by takeit**
See the README.md file for the documentation.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+matrix:
+  include:
+    - php: 5.4
+    - php: 5.5
+    - php: 5.6
+    - php: hhvm
+    - php: nightly
+    - php: 7.0
+      env: COMPOSER_FLAGS='--prefer-lowest --prefer-stable'
```
Comment (ahilles107) [inline .travis.yml]: this same as in bundle

File: Resolver/TenantResolver.php [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    public function resolve($host = null)
+    {
+        // fallback to default tenant
+        if (null === $host) {
+            $host = self::DEFAULT_TENANT;
+        }
+
+        $subdomain = $this->extractSubdomain($host);
+
+        return $this->tenantRepository->findBySubdomain($subdomain);
```
Comment (takeit) [inline Resolver/TenantResolver.php]: should not make extra query when the tenant is default. It should return populated (with default tenant data) object of type Tenant

File: Context/TenantContext.php [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+     */
+    protected $tenant;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getTenant()
+    {
+        if (null === $this->tenant) {
+            return new Tenant();
```
Comment (takeit) [inline Context/TenantContext.php]: It should return populated (with default tenant data) object of type Tenant

**Comment by ahilles107** [comment]
:+1:

**Pull request merged**